### PR TITLE
blog: preview environments

### PR DESCRIPTION
**Pull request by nicks**
Hello @lianmakesthings, @milas, @nicksieger,

Please review the following commits I made in branch nicks/ephemerator:

beea2a8c50407c9e9bb0e6574dff3ba989d55ef8 (2022-02-14 21:55:20 -0500)
blog: preview environments

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

**Comment by nicks** [comment]
this is a write-up of what was kind of a bizarre experiment that i don't really want to spend more time on.

but i do think that the experiment gave me a lot of ideas on how we should improve tilt for multi-service apps, particularly around doing subdomains and auth and cluster setup locally

File: blog/_posts/2022-02-14-preview-environments.md [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+PR. And this seems to be becoming table stakes in Jamstack platforms like
+Netlify and Vercel.
+
+### Cost Control
+
+A more mature preview env platform would destroy or scale down envs that aren't
+  being used [to save on
+  cost](https://ephemeralenvironments.io/features/cost-control/).
+  
+The ephmerator is very native: every env has a 15 minute expiry. Then it's destroyed.
```
Comment (nicksieger) [inline blog/_posts/2022-02-14-preview-environments.md:245]: I think this is what you meant here
```suggestion
The ephmerator is very naïve: every env has a 15 minute expiry. Then it's destroyed.
```

**Review by nicksieger — APPROVED** [review]
Love the origin story of the crazy experiment! I do wish that all the individual pieces you had to solve for didn't have such sharp edges. Looking forward to seeing where this goes.

File: blog/_posts/2022-02-14-preview-environments.md [hunk 2]
```diff
@@ -0,0 +1,322 @@
+---
+slug: "preview-environments"
+date: 2022-02-14
+author: nick
+layout: blog
+title: "How I Built a Toy Preview Environment Platform and What Happened Next"
```
Comment (lianmakesthings) [inline blog/_posts/2022-02-14-preview-environments.md:6]: This title is really a mouthful, plus it's almost completely duplicated in the subtitle.
Suggest to make it slightly more lightweight
I tried building / built a preview environment platform. Here's what happened

**Review by lianmakesthings — APPROVED** [review]

File: blog/_posts/2022-02-14-preview-environments.md (hunk 2 above)
Comment (nicks) [reply blog/_posts/2022-02-14-preview-environments.md:6]: ya, i totally agree. want to think about a more troll-y title. :thinking: :thinking: :thinking:

Comment (nicksieger) [reply blog/_posts/2022-02-14-preview-environments.md:6]: A Preview Environment Preview: The Operator Strikes Back

Operators Previewing Preview Environments Preview Preview Environments

Comment (milas) [reply blog/_posts/2022-02-14-preview-environments.md:6]: Experiments With Preview Environments (And Making Security Professionals Cringe)

Preview Environments As Dev Environments As Code

Experiments in `kink`: Using Kubernetes-in-Kubernetes To Create Preview Environments

**Review by milas — APPROVED** [review]

**Pull request merged**